text/0000-assoc-default-groups.md

```diff
@@ -0,0 +1,1733 @@
+available for specialization, the `$default_type` may be assumed
+in other items specified in the implementation.
+If an implementation does make the associated type available for
+further specialization, then other definitions in the implementation
+may not assume the given underlying specified type of the associated type
+and may only assume that it is `Self::TheAsociatedType`.
+
+This applies generally to any item inside a `trait`.
+You may only assume the signature of an item, but not any provided definition,
+in provided definitions of other items.
+For example, this means that you may not assume the value of an
+associated `const` item in other items with provided definition
+in a `trait` definition.
+
+### Interaction with `dyn Trait<...>`
+
++ Let `σ` denote a well-formed type.
++ Let `L` denote a well-formed lifetime.
++ Let `X` refer to an object safe `trait`.
+  + Let `k` denote the number of lifetime parameters in `X`.
+  + Let `l` denote the number of type parameters in `X`.
+  + Let `m` where `0 ≤ m ≤ l` denote the number of type parameters
+    in `X` without specified defaults.
+  + Let `A` denote the set of associated types in `X`.
+  + Let `o = |A|`.
+  + Let `D` where `D ⊆ A` denote set of associated types in `X` with defaults.
+  + Let `E = A \ D`.
+
+Then, in a type of form (where `m ≤ n ≤ l`):
+
+```rust
+dyn X<
+  L0, .., Lk,
+  σ0, .. σn,
+  A0 = σ_{n + 1}, .., Ao = σ_{n + o}
+>
+```
+
+the associated types in `E` must be bound in `A0, .., Ao`
+whereas those in `D` may be omitted selectively (i.e. omit zero, some, or all).
+
+When inferring the types of the omitted projections in `D`,
+projections in the assigned defaults of types in `D` will use the types in
+`A0, .., Ao` instead of the defaults specified in `D`. For example, if given:
+
+```rust
+trait X {
+    type A0 = u8;
+    type A1 = Vec<Self::A0>;
+}
+```
+
+then the type `dyn X<A0 = u16>` is inferred to `dyn X<A0 = u16, A1 = Vec<u16>>`
+as opposed to `dyn X<A0 = u16, A1 = Vec<u8>>`.
+
+### Interaction with `existential type`
+
+[RFC 2071]: https://github.com/rust-lang/rfcs/blob/master/text/2071-impl-trait-existential-types.md#reference-existential-types
+
+[RFC 2071] defines a construct `existential type Foo: Bar;` which is permitted
+in associated types and results in an opaque type. This means that the nominal
+type identity is hidden from certain contexts and only `Bar` is extensionally
+known about the type wherefore only the operations of `Bar` is afforded.
+This construct is sometimes written as `type Foo = impl Bar;` in conversation
+instead.
+
+[RFC 1210]: https://github.com/rust-lang/rfcs/blob/master/text/1210-impl-specialization.md#default-impls
+
+With respect to this RFC, the semantics of `type Assoc = impl Bar;`
+inside a trait definition, where `Assoc` is the name of the associated type,
+is understood as what it means in terms of `default impl ..` as discussed
+in [RFC 1210]. What this entails means in concrete terms is that given:
```

Comment:
typo: you should remove either the word "entails" above or the phrase "means in concrete terms"